src/workflows/runtime/broker.py

```diff
@@ -162,11 +161,7 @@ async def _run_workflow() -> None:
                     finally:
                         # ensure run context is cleaned up even on failure
                         workflow_registry.delete_run(run_id)
-                    result.set_result(
-                        workflow_result.result
-                        if type(workflow_result) is StopEvent
-                        else workflow_result
-                    )
+                    result.set_result(workflow_result)
```

Comment:
no, this is a big breaking change. This sets the result of the `WorkflowHandler` future. The value here is the result of `await workflow.run()`